pandas/sparse/scipy_sparse.py

```diff
@@ -0,0 +1,102 @@
+"""
+Interaction with scipy.sparse matrices.
+
+Currently only includes SparseSeries.to_coo helpers.
+"""
+from pandas.core.frame import DataFrame
+from pandas.core.index import MultiIndex, Index
+from pandas.core.series import Series
+import itertools
+import numpy
+from pandas.compat import OrderedDict
+from pandas.tools.util import cartesian_product
+
+
+def _get_label_to_i_dict(labels, sort_labels=False):
+    """ Return OrderedDict of unique labels to number. Optionally sort by label. """
+    labels = Index(map(tuple, labels)).unique().tolist()  # squish
```

Comment:
I would consider just returning an MultiIndex from this function instead of an OrderedDict. The multi-index supports the dictionary queries you need with `get_loc`/`get_indexer` and will be much faster. Also, you might be able to simply/speedup the logic considerably by using pandas methods:
```
labels = Index(labels).drop_duplciates()
if sort_labels:
    labels = labels.order()
return labels
```